library/lua/dfhack.lua

```diff
@@ -375,21 +408,38 @@ function xy2pos(x,y)
     end
 end
 
+---@nodiscard
+---@param a coord|coord2d
+---@param b coord|coord2d
+---@return boolean
 function same_xy(a,b)
     return a and b and a.x == b.x and a.y == b.y
 end
 
+---@nodiscard
+---@param path coord_path|coord2d_path
+---@param i number
+---@return integer x
+---@return integer y
 function get_path_xy(path,i)
     return path.x[i], path.y[i]
 end
 
+-- Walks a sequence of dereferences, which may be represented by numbers or
+-- strings. Returns nil if any of obj or indices is nil, or a numeric index is
+-- out of array bounds.
+---@param obj table
+---@param idx number|string
+---@param ... number|string
+---@return any obj
```

Comment:
I'm not fully aware of what the syntax is for the annotations, but should this be "any" instead of "any obj"?

Reply:
Return types in LuaLS follow this syntax:
`---@return <type> [<name> [comment] | [name] #<comment>]`